### PR TITLE
Some fixes for view duration and added tests

### PR DESCRIPTION
**Pull request by ijunaid**

### CODE REVIEW

File: CountlyTests/CountlyViewTests.swift [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        let resumeViewExpectation = XCTestExpectation(description: "Wait for resume view")
+        DispatchQueue.global().asyncAfter(deadline: .now() + 4) {
+            Countly.sharedInstance().views().resumeView(withID: viewID)
+            resumeViewExpectation.fulfill()
+        }
+        
+        wait(for: [pauseViewExpectation, resumeViewExpectation], timeout: 10)
+    }
+    
+    func testViewTrackingWithBackgroundAndForegroundNotifications() throws {
```
Comment (arifBurakDemiray) [inline CountlyTests/CountlyViewTests.swift:178]: addition of test comments would be great

**Review by arifBurakDemiray — APPROVED** [review]

File: CountlyTests/CountlyViewTests.swift [hunk 2]
```diff
@@ -10,190 +10,506 @@ import XCTest
...
+        XCTAssertNotNil(viewID, "View should be started successfully.")
+        
+        let expectation = XCTestExpectation(description: "Wait for 3 seconds before stopping the view.")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
+            Countly.sharedInstance().views().stopView(withName: "View1")
+            expectation.fulfill()
         }
         
+        wait(for: [expectation], timeout: 5.0) // Wait for the expectation to be fulfilled
+        checkRecordedEventsForView(viewName: "View1")
```
Comment (arifBurakDemiray) [inline CountlyTests/CountlyViewTests.swift:27]: you may also want to validate duration of the views

File: CountlyTests/CountlyViewTests.swift [hunk 3]
```diff
@@ -10,190 +10,506 @@ import XCTest
...
+            expectation.fulfill()
         }
         
-        let resumeViewExpectation = XCTestExpectation(description: "Wait for resume view")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 10) { // Delayed by 10 seconds
-            Countly.sharedInstance().views().resumeView(withID: viewID)
-            resumeViewExpectation.fulfill()
-        }
+        wait(for: [expectation], timeout: 5.0) // Wait for the expectation to be fulfilled
+        checkRecordedEventsForView(viewName: "View1", segmentation: ["key": "value"])
```
Comment (arifBurakDemiray) [inline CountlyTests/CountlyViewTests.swift:44]: for example here. View must have a duration of 4 secondo

File: CountlyTests/CountlyViewTests.swift [hunk 4]
```diff
@@ -10,190 +10,506 @@ import XCTest
...
+            Countly.sharedInstance().views().stopView(withID: viewID)
+            expectation.fulfill()
         }
         
-        let fgExpectation = XCTestExpectation(description: "Wait for active notification")
-        DispatchQueue.global().asyncAfter(deadline: .now() + 20) { // Delayed by 20 seconds
-            NotificationCenter.default.post(name: UIApplication.didBecomeActiveNotification, object: nil)
-            fgExpectation.fulfill()
+        wait(for: [expectation], timeout: 5.0) // Wait for the expectation to be fulfilled
+        checkRecordedEventsForView(withID: viewID)
```
Comment (arifBurakDemiray) [inline CountlyTests/CountlyViewTests.swift:61]: we want to see specifically also those durations are validated and SDK correctly calculates the duration

**Review by arifBurakDemiray — CHANGES_REQUESTED** [review]
This is a nice addition of test suite.
It might be extended by scenarios
For example what happens after consent removal etc.

**Review by AliRKat — APPROVED** [review]
cool work! looks all good to me. just couldn't be sure whether we validate each param under a view request or not but I don't think there's anything wrong with the current way

**Review by turtledreams — APPROVED** [review]

**Pull request merged**